Add article HTML full text hide/show in record view

camillevilla opened the pull request:
Closes #1509 

This PR adds hide/show functionality for HTML fulltext in the article record view:
- Hides full text by default
- Clicking on the section header (green bar) toggles fulltext
- `View full text / Show full text` does not appear in records where HTML full text is unavailable
- adds a concern for `EdsDocument`

## Demo
![hide-show-fulltext](https://user-images.githubusercontent.com/5402927/28686780-44a28738-72c1-11e7-97d2-fec219879f84.gif)

jvine commented:
@camillevilla I like the clickable bar! Nice. The animation looks good but I'd like to see the HTML in action - I can't quite tell from the code what's happening w ARIA etc. Screenhero at some point? 

Couple of clean-up bits for the display:

- Change Full Text -> Full text  
(SUL style is sentence case)
- remove the `<dl>` formatting for the full text section, so it doesn't have the redundant label to the left.

Current
![screen shot 2017-07-25 at 10 22 42 am](https://user-images.githubusercontent.com/6945691/28585071-ed20ce82-7123-11e7-9c54-b6e928509701.png)

Mockup
![screen shot 2017-07-25 at 10 25 34 am](https://user-images.githubusercontent.com/6945691/28585127-1c41ac54-7124-11e7-84d4-179bb39e109a.png)

camillevilla commented:
hi @jvine! I've made the following updates:
- moved fulltext out of `<dl>`
- updated section header to sentence case
- renamed our div ID / `aria-control` to `toggleFulltext`

camillevilla commented:
@jkeck or @drh-stanford just did a walkthrough with @jvine and got the thumbs up. I had to

jkeck commented on app/helpers/article_helper.rb:
```diff
@@ -11,6 +11,10 @@ def article_restricted?(document)
   end
 
+  def html_fulltext_available?(document)
+    !document['eds_html_fulltext'].blank?
```
What do you think about this a method on the `SolrDocument` class instead of a helper that gets passed an instance of a document?

jkeck commented on app/views/article/record/_html_fulltext.erb:
```diff
@@ -0,0 +1,13 @@
+<% doc_presenter = presenter(document) -%>
+<div class="section-container-heading" data-toggle="collapse" href="#toggleFulltext" aria-expanded="false" aria-controls="toggleFulltext">
+  <h2>Full text</h2>
+</div>
```
I haven't checked this branch out yet (in the middle of a bunch of changes in my working branch at the moment), but a couple questions I have:

Is this `div` keyboard accessible (e.g. can you tab to it and hit enter to open/close)?  Does the cursor indicate that the heading element is clickable (it didn't look like it from the gif, but can be hard to tell).

camillevilla replied:
@jvine and I talked about this a little bit yesterday. The `div` itself is not tab-able, but the second link (Hide / View full text) is.

The cursor does not indicate the `div` is clickable. Maybe I could turn it into a giant button?

jvine commented:
Just to note that turning the toggle bar into a button will turn it into a duplicate control for keyboard and SR users, which is not necessarily the goal. I'm feeling like it might be better to turn the toggle into the sole control, and make it explicit. That would eliminate the sidebar box when there's no PDF version, but that's probably fine.

jkeck commented on app/views/article/record/_html_fulltext.erb:
```diff
@@ -0,0 +1,13 @@
+  <% field_name = field_name.to_s -%>
+  <% value = doc_presenter.field_value field_name -%>
+  <% if value.present? %>
+    <dd class="blacklight-<%= field_name.parameterize %>"><%= value %></dd>
```
It looks like this is putting the content in a `dd` w/o any of the other markup (parent `dl` and an accompanying `dt`).

Check out the [Permitted parents section](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dd) of this page.

I think we can just wrap this in a `div` and call it a day.

jkeck commented on app/assets/stylesheets/modules/record-metadata-section.scss:
```diff
@@ -1,40 +1,37 @@
+  }
+
+  .section-container-heading {
+    background-color: #009B76;
```
Can we either use an existing scss variable or turn this into one?

camillevilla commented:
alrighty, @jkeck! Here are the requested changes.
